### PR TITLE
feat: implement transaction context

### PR DESCRIPTION
**Pull request by lukas-reining**
Implements transaction context as described in #292.
This is done the same as in Node, the callback we have for Node is not needed with context vars, like it is in Java.

@federicobond just after I did the changes I saw you PR from May:
Sorry, I could have just continued test on this one. 
We basically did the same but I added some tests also for the context merging.

### CODE REVIEW

**Review by beeme1mr — APPROVED** [review]

File: openfeature/transaction_context/__init__.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+from openfeature.transaction_context.no_op_transaction_context_propagator import (
+    NoOpTransactionContextPropagator,
+)
+from openfeature.transaction_context.transaction_context_propagator import (
+    TransactionContextPropagator,
+)
+
+__all__ = [
+    "TransactionContextPropagator",
+    "NoOpTransactionContextPropagator",
```
Comment (gruebel) [inline openfeature/transaction_context/__init__.py:13]: to keep it aligned with the other `NoOp` classes I would prefer not to expose it here and when needed just import it via the `no_op_transaction_...` file directly.

File: openfeature/transaction_context/no_op_transaction_context_propagator.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+)
+
+
+class NoOpTransactionContextPropagator(TransactionContextPropagator):
+    def get_transaction_context(self) -> EvaluationContext:
+        return EvaluationContext()
+
+    def set_transaction_context(
+        self, transaction_context: EvaluationContext
+    ) -> None: ...
```
Comment (gruebel) [inline openfeature/transaction_context/no_op_transaction_context_propagator.py:13]: also here to be consistent with the code, just use `pass`
```suggestion
    ) -> None:
    pass
```

`...` in `Protocol` is correct 🙂

File: openfeature/transaction_context/transaction_context_propagator.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import typing
+from typing import TypeVar
+
+from openfeature.evaluation_context import EvaluationContext
+
+T = TypeVar("T", bound="TransactionContextPropagator")
```
Comment (gruebel) [inline openfeature/transaction_context/transaction_context_propagator.py:6]: Maybe you planed to somehow use it, but I don't see anything. please elaborate or remove it.

**Review by gruebel — COMMENTED** [review]
nice work, just added a few minor comments

File: openfeature/transaction_context/transaction_context_propagator.py (hunk 3 above)
Comment (lukas-reining) [reply openfeature/transaction_context/transaction_context_propagator.py]: Yes it was from before, thanks!

**Review by gruebel — APPROVED** [review]

**Comment by gruebel** [comment]
I'll give @federicobond till end of week to leave some comments, otherwise will merge this and close the previous draft.

**Pull request merged**